frontend/rolecall/src/app/user/user-editor.component.ts

```diff
@@ -291,42 +296,44 @@ export class UserEditor implements OnInit {
     if (info.type == "date") {
       let date = Date.parse(val.value);
       val = date;
-      this.userSaved = false;
+      this.disableSave = false;
     }
     else if (info.type == "permissions") {
       let val2 = this.workingUser.has_permissions;
       for (let entry of Object.entries(val2)) {
         if (val.includes(entry[0])) {
           if (!val2[entry[0]]) {
-            this.userSaved = false;
+            this.disableSave = false;
           }
           val2[entry[0]] = true;
         } else {
           if (val2[entry[0]]) {
-            this.userSaved = false;
+            this.disableSave = false;
           }
           val2[entry[0]] = false;
         }
       }
       val = val2;
-    }
-    else if (key == "Privilege Classes") {
-      if (this.workingUser.has_privilege_classes) {
-        let largerArr = val.length > this.workingUser.has_privilege_classes.length ? val : this.workingUser.has_privilege_classes;
-        let smallerArr = val.length <= this.workingUser.has_privilege_classes.length ? val : this.workingUser.has_privilege_classes;
-        for (let i = 0; i < largerArr.length; i++) {
-          if (largerArr[i] != smallerArr[i]) {
-            this.userSaved = false;
-            break;
+    } else if (info.type == "roles") {
+      let val2 = this.workingUser.has_roles;
```

Comment:
1. Let's avoid variable names like "val" and "val2" where possible and name it something more semantic
2. Let's make this a const since it is not reassigned
3. `has_roles` makes me expect a boolean true/false, but it looks like it contains a list? Seems surprising.

Reply:
Agree 100%. However, fixing it is not trivial.  As I mentioned above, I decided to copy the functionality of permissions, that already worked fine, when I fixed the broke role system. The code at line 317 is identical to the code at 311 except 311 pertains to Permissions and 317 pertains to Roles.  All I did when I added this code was to change Permissions to Roles and to verify that all variables I used were local. I was right, the code works, and because it is logically identical, it should make the program more difficult to understand. On the down-side, I also inherited whatever bad programming practices were used before.

Reply:
Acknowledging that we are inheriting a lot of this stuff, and we don't have to fix everything all at once.

Reply:
in my earlier comment, I meant "... it should NOT make the program more difficult to understand."